go/sig3/prot.go

```diff
@@ -0,0 +1,133 @@
+package sig3
+
+import (
+	keybase1 "github.com/keybase/client/go/protocol/keybase1"
+)
+
+type UID [16]byte
+type LinkType int
+type ChainType = keybase1.SeqType
+type SigVersion int
+type LinkID [32]byte
+type Seqno = keybase1.Seqno
+type Time = keybase1.Time
+type IgnoreIfUnsupported bool
+type KID = keybase1.BinaryKID
+type TeamID [16]byte
+type PerTeamKeyGeneration = keybase1.PerTeamKeyGeneration
+type Entropy []byte
+type Sig [64]byte
+type PTKType int
+
+const (
+	SigVersion3 SigVersion = 3
+)
+
+// These values are picked so they don't conflict with Sigchain V1 and V2 link types
+const (
+	LinkTypeNone      LinkType = 0
+	LinkTypeUserPeg   LinkType = 65
+	LinkTypeRotateKey LinkType = 81
+)
+
+// The values are picked so they don't conflict with Sigchain V1 and V2 SeqType's
+const (
+	ChainTypeUserPrivateHidden ChainType = 16
+	ChainTypeTeamPrivateHidden ChainType = 17
+)
+
+const (
+	// The default, legacy type. All users, including the readers, can access it.
+	PTKTypeReader PTKType = 0
+)
+
+// OuterLink V3 is the third version of Keybase sigchain signatures, it roughly approximates
+// the outer link v2s that we have previously used.
+type OuterLink struct {
+	_struct             bool                `codec:",toarray"`
+	Version             SigVersion          `codec:"version"` // should be 3
+	Seqno               Seqno               `codec:"seqno"`
+	Prev                *LinkID             `codec:"prev"`
+	InnerLink           LinkID              `codec:"curr"` // hash of the msgpack of the InnerLink
+	LinkType            LinkType            `codec:"type"` // hash of the msgpack of the previous OuterLink
+	ChainType           ChainType           `codec:"chaintype"`
+	IgnoreIfUnsupported IgnoreIfUnsupported `codec:"ignore_if_unsupported"`
+	// New field for V3; if this link is encrypted, specify the format, nonce and PUK
+	EncryptionParameters *EncryptionParameters `codec:"encryption_parameters"`
```

Comment:
I don't see this field in proofs/src/sig3.iced?

Reply:
good point, it shoudl be there

Reply:
https://github.com/keybase/proofs/pull/143/files